cloudevents/sdk/http_events.py

```diff
@@ -0,0 +1,217 @@
+    Currently only supports binary events
+    """
+
+    def __init__(
+            self,
+            headers: dict,
+            data: any,
+            binary: bool = True,
+            f: typing.Callable = lambda x: x
+    ):
+        """
+        Event HTTP Constructor
+        :param headers: a dict with cloudevent specified metadata
+            e.g. {
+                "content-type": "application/cloudevents+json",
+                "ce-id": "16fb5f0b-211e-1102-3dfe-ea6e2806f124",
+                "ce-source": "<event-source>",
+                "ce-type": "cloudevent.event.type",
+                "ce-specversion": "0.2"
+            }
+        :type headers: dict
+        :param data: a data object to be stored inside Event
+        :type data: any
+        :param binary: a bool indicating binary events
+        :type binary: bool
+        :param f: callable function for reading/extracting data
+        :type f: typing.Callable
+        """
+        headers = {key.lower(): headers[key] for key in headers}
+        if binary:
+            for field in base._ce_required_fields:
+                if field not in headers:
+                    raise TypeError(
+                        "parameter headers has no required attribute {0}"
+                        .format(
+                            field
+                        ))
+
+                if not isinstance(headers[field], str):
+                    raise TypeError(
+                        "in parameter headers attribute "
+                        "{0} expected type str but found type {1}".format(
+                            field, type(headers[field])
+                        ))
+
+            for field in base._ce_optional_fields:
+                if field in headers and not isinstance(headers[field], str):
+                    raise TypeError(
+                        "in parameter headers attribute "
+                        "{0} expected type str but found type {1}".format(
+                            field, type(headers[field])
+                        ))
+        else:
+            raise Exception("not implemented")
+        self.headers = headers
+        self.data = data
+        self.marshall = marshaller.NewDefaultHTTPMarshaller()
+        self.event_handler = v1.Event()
+        self.marshall.FromRequest(
+            self.event_handler,
+            self.headers,
+            self.data,
+            f
+        )
+
+    def emit(
+        self,
+        url: str,
+        binary: bool = True,
+        method: typing.Callable = requests.post
+    ) -> any:
+        """
+        Sends HTTP POST event to given url
+
+        :param url: a string referencing target to send event to
+        :type url: str
+        :param binary: a bool indicating whether this is a binary event
+        :type binary: bool
+        """
+        if binary:
+            return self.emit_binary_event(url, method=method)
+        else:
+            return self.emit_structured_event(url)
```

Comment:
I would not expect this library to contain support for emitting events, and I don't believe the other SDKs contain similar logic. I think we should probably leave this for the user to handle.

Reply:
Removed emitting events, we'll discuss this later in a design doc